api/axis.py

```diff
@@ -1253,6 +1295,22 @@ def _raise_exception_on_error(cls, code, message, custom_error_classes={}):
         return code, message
 
 
+class CheckinResponseParser(ResponseParser):
+
+    def process_all(self, string):
+        for i in super(CheckinResponseParser, self).process_all(
+                string, "//axis:EarlyCheckinRestResult", self.NS):
+            return i
+
+    def process_one(self, e, namespaces):
+        """Either raise an appropriate exception, or do nothing.
+        """
+        self.raise_exception_on_error(
+            e, namespaces, ignore_error_codes = [4058]
```

Comment:
Most of the time, "No checkout is associated with patron for the title" is an error. But when returning a book, we can just claim that the return has succeeded. This makes the return operation idempotent.